### PR TITLE
fix unresolved dartdoc links

### PR DESCRIPTION
**Pull request by devoncarew**
Fix unresolved dartdoc links:
- expand relative links in dartdoc comments to absolute links
- convert `[FontFace/status]` member references in dartdoc comments to `[FontFace.status]` ones

Starting as a draft (so https://github.com/dart-lang/web/pull/198/ can land first).

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]
Amazing, thanks!

File: tool/scrape_mdn.dart [hunk 1]
```diff
@@ -166,6 +173,24 @@ String convertMdnToMarkdown(String content) {
     lines.removeLast();
   }
 
+  // Rewrite relative link references:
+  //   "[WebGL API](/en-US/docs/Web/API/WebGL_API)"
+  final linkRefRegex = RegExp(r'\[([^\]]+)\]\(([\w\/-]+)\)');
+  // ignore: prefer_expression_function_bodies
```
Comment (srujzs) [inline tool/scrape_mdn.dart:179]: I'm guessing the formatter does some weird wrapping if we make this an expression instead? Maybe add a comment on why we ignore the lint.

**Review by srujzs — APPROVED** [review]
Awesome, thanks Devon!

File: tool/scrape_mdn.dart (hunk 1 above)
Comment (devoncarew) [reply tool/scrape_mdn.dart:179]: It's not terrible (the formatting) - I just tend not to use the shorthand syntax for functions that are longer than a line or two:

Screenshot 2024-03-12 at 12 38 29 PM

Comment (devoncarew) [reply tool/scrape_mdn.dart]: (removed the todo and re-formatted the expression)

Comment (srujzs) [reply tool/scrape_mdn.dart]: Yeah I also tend to avoid the expression syntax for long functions. Either way sounds fine to me, thanks!

**Pull request merged**